src/v/security/audit/schemas/hashing_utils.cc

```diff
@@ -0,0 +1,237 @@
+      class_uid::api_activity,
+      severity_id::informational,
+      crud);
+    /// actor
+    boost::hash_combine(h, actor_hash(auth_result));
+    /// api
+    boost::hash_combine(h, operation_name);
+    boost::hash_combine(h, service_name);
+    /// dst_endpoint
+    boost::hash_combine(
+      h, std::hash<ss::net::inet_address>{}(local_address.addr()));
+    /// src_endpoint
+    boost::hash_combine(h, std::hash<ss::net::inet_address>{}(client_addr));
+    /// status_id
+    boost::hash_combine(h, auth_result.authorized);
+    /// unmapped_data
+    if (auth_result.acl.has_value()) {
+        boost::hash_combine(
+          h, absl::Hash<security::acl_entry>{}(*auth_result.acl));
+    }
+    if (auth_result.resource_pattern.has_value()) {
+        boost::hash_combine(
+          h,
+          absl::Hash<security::resource_pattern>{}(
+            *auth_result.resource_pattern));
+    }
+    return h;
+}
+
+size_t http_request_hash(const ss::http::request& req) {
+    size_t h = 0;
+    /// request headers
+    for (const auto& hdr : req._headers) {
+        boost::hash_combine(h, hdr.first);
+        boost::hash_combine(h, hdr.second);
+    }
+    /// request method
+    boost::hash_combine(h, req._method);
+    /// url
+    boost::hash_combine(h, req._url);
+    boost::hash_combine(
+      h, std::hash<ss::net::inet_address>{}(req.get_server_address().addr()));
+    boost::hash_combine(h, req.get_protocol_name());
+    boost::hash_combine(h, req.get_url());
+    // version
+    boost::hash_combine(h, req._version);
+    return h;
+}
+
+size_t api_activity::hash(
+  ss::httpd::const_req req,
+  const request_auth_result& auth_result,
+  const ss::sstring& svc_name,
+  bool authorized,
+  const std::optional<std::string_view>& reason) {
+    const auto activity_id = http_method_to_activity_id(req._method);
+    size_t h = audit_type_base_hash(
+      category_uid::application_activity,
+      class_uid::api_activity,
+      severity_id::informational,
+      activity_id);
+    /// actor
+    ///  -- authorization_result
+    ///    -- policy
+    boost::hash_combine(h, auth_result.is_auth_required());
+    if (reason) {
+        boost::hash_combine(h, reason);
+    }
+    boost::hash_combine(h, svc_name);
+    ///  -- user
+    boost::hash_combine(h, auth_result.get_username());
+    boost::hash_combine(h, auth_result.is_superuser());
+    /// dst_endpoint
+    boost::hash_combine(
+      h, std::hash<ss::net::inet_address>{}(req.get_server_address().addr()));
+    /// http_request
+    boost::hash_combine(h, http_request_hash(req));
+    /// src_endpoint
+    boost::hash_combine(
+      h, std::hash<ss::net::inet_address>{}(req.get_client_address().addr()));
+    /// status_id
+    boost::hash_combine(h, authorized);
+    return 0;
+}
+
+size_t api_activity::hash(
+  ss::httpd::const_req req,
+  const ss::sstring& user,
+  const ss::sstring& svc_name) {
+    const auto activity_id = http_method_to_activity_id(req._method);
```

Comment:
@michael-redpanda do we need the activity id to calculate the base hash? All that would matter is hashing the method I think. Then we could make this method private to utils.h again

Reply:
This I believe is for the API activity, yes?  API activity includes both HTTP and Kafka.  Kafka doesn't have a 'method' field.  So yes, I believe we do need to use activity ID in the hash.  You should double check, but does "type_id" of the base type get hashed?  If so then "activity_id" is already being hashed as Type ID is a combination of class ID and activity ID (https://schema.ocsf.io/1.0.0/classes/api_activity?extensions= look towards the bottom).

Reply:
Yeah for sure, i was just wondering if it would be good enough to has something like 0 - 5, one value for each of the potential HTTP method types, instead of having to get one of 5 stringified versions of the enum to later just hash